recipes/nasm/all/conanfile.py

```diff
@@ -105,3 +105,11 @@ def package(self):
     def package_info(self):
         self.cpp_info.libdirs = []
         self.cpp_info.includedirs = []
+
+        bin_folder = os.path.join(self.package_folder, "bin")
+        # In case need to find packaged tools when building a package
+        self.buildenv_info.append("PATH", bin_folder)
+        # In case need to find packaged tools at runtime
+        self.runenv_info.append("PATH", bin_folder)
```

Comment:
```suggestion
```

no, useless

Reply:
Hi @SpaceIm  - in the interest of ensuring that this community remains a positive environment, I would like to point out that we assume that code contributors have done it to the best of their abilities, and we expect reviewers to provide feedback in a constructive way. Likewise, we expect contributors to accept constructive feedback gracefully. 

It would perhaps be beneficial, to both the author and anyone else who revisits this PR, to provide more context as to why these lines are not needed.

Reply:
It's useless & broken:

- useless because `PATH` is a first class citizen in conan v2 helpers (bindirs of tool_requires recipes automatically populates `PATH`).
- broken because we don't add path values to `self.buildenv_info` & `self.runenv_info` with `append`. They have dedicated methods for paths: `define_path`, `prepend_path`, `append_path`.

Reply:
Yes thanks spaceim and jcar87, I'm doing my best and I appreciate your time and feedback.  I just want things to work so I can move onto using this nice system, but I have limited time to do a bunch of these upgrades, so I'm pushing as hard as I can. Things will be missed and misunderstood.

I'm going to leave this unresolved so the discussion isn't hidden,

I try to follow what I see in templates and other upgraded recipes, so there will be other recipes that have the same pattern - perhaps something to add to the linter.

I'm still not clear on the whole "first class citizen" expression you are using here (I don't know the significance of being first or second class in this context), and what should be in the recipe.

Is this correct ? :
- For version 1 recipes, we need `self.env_info.PATH.append(bin_path)`, until everything is upgraded to v2.
- For version 2 recipes, we don't need to add anything here at all (?) because the bindirs will be automatically added into the environments of recipes that have declared VirtualBuildEnv or VirtualRunEnv.
- If I wanted to add any extra paths (additional to the standard binpath) to the environment of the _consumers_ (?) of a recipe, then I call things like self.buildenv_info.prepend_path(something)